src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+package parser;
+
+import java.util.ArrayList;
+
+import exception.DukeException;
+import exception.InvalidInputException;
+import exception.NoTaskDescriptionException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import tasklist.TaskList;
+import ui.Ui;
+
+/**
+ * Class to encapsulate all parser methods, which parse through user inputs.
+ */
+public class Parser {
+    /**
+     * Returns Task formed from given string.
```

Comment:
Perhaps another verb other than "Returns" would be better for JavaDoc comments, as per the coding standard.